docs/src/proposals/append-vec-storage.md

```diff
@@ -0,0 +1,231 @@
+The account pubkey is accessed via the `local_id` of the `StoredAccountMetaV2`,
+which is derived from the location of the account meta when an AppendVec is
+first opened.  Specifically, one account pubkey is located at
+`account_pubkey_block_offset` + 32 + `local_id` * `ACCOUNT_PUBKEY_SIZE`.
+
+```
++--------------------------------------+
+| Hash (32 bytes for the entire block) |
++--------------------------------------+
+| account pubkey 1                     |
++--------------------------------------+
+| account pubkey 2                     |
++--------------------------------------+
+|         ...                          |
++--------------------------------------+
+| account pubkey N                     |
++--------------------------------------+
+```
+
+```
++-------------------------------------------------------------+
+| Account Pubkey (32 bytes each)                              |
++-------------------------------------------------------------+
+| pubkey          | 32 bytes                                  |
++-------------------------------------------------------------+
+```
+
+#### Owners Block
+The owners block stores the owner's information.  An owner entry can be accessed
+via `owner_block_offset` + 32 + `owner_local_id` * `OWNER_ENTRY_SIZE`.
+
+```
++--------------------------------------+
+| Hash (32 bytes for the entire block) |
++--------------------------------------+
+| owner 1                              |
++--------------------------------------+
+| owner 2                              |
++--------------------------------------+
+|         ...                          |
++--------------------------------------+
+| owner N                              |
++--------------------------------------+
+```
+
+```
++-------------------------------------------------------------+
+| Owner (40 bytes each)                                       |
++-------------------------------------------------------------+
+| pubkey          | 32 bytes                                  |
+| write version   |  8 bytes                                  |
++-------------------------------------------------------------+
+```
+
+#### Account Data Blocks
+
+
+```
++------------------------------------+
+| Data blocks hash | 32 bytes        |
++------------------------------------+
+| Account data 1                     |
++------------------------------------+
+| Account data 2                     |
++------------------------------------+
+| Account data 3                     |
++------------------------------------+
+|          ...                       |
++------------------------------------+
+| Account data N                     |
++------------------------------------+
+```
+
+```
++-------------------------------------------------------------+
+| Account data block (variable size)                          |
++-------------------------------------------------------------+
+| rent_epoch      |  0-8 bytes                                |
```

Comment:
I think this may have to be aligned.

Reply:
Feel the same way.  Then we might need to store the account data size for each account as there might be gaps between two account data entries with paddings.

Reply:
How is this accessed? I imagine it's read, not directly mmapped, so it can be left unaligned and read with https://doc.rust-lang.org/stable/std/ptr/fn.read_unaligned.html